README.md

```diff
@@ -0,0 +1,88 @@
+```sh
+  ################################################
+  ##              ###                            ##
+  ##             ##                              ##
+  #######       ##      ##       ##     ###     ##
+      ##       ##        #########     ##      ##
+     ##       ##               ##           ###
+    ##        #######         ##           ##
+   ##               ##       ##             ##
+  ##                       ###      ###      ##
+  ##################################  #########
+
+     Z S E R I O  C + +  R E F L E C T I O N
+```
+
+# About
+
+Zsr provides a c++ runtime reflection layer for [zserio](https://github.com/ndsev/zserio) generated c++ modules.
+
+# Building
+
+```sh
+mkdir build && cd build
+cmake -DZSERIO_RT_DIR=<path/to/zserio/compiler/extensions/cpp/runtime/src> ..
+cmake --build .
+```
+
+```sh
+ctest .
+```
+
+# Usage
+
+## CMake
+
+In your CMake project, call the `add_zserio_module` function to generate and
+build a zserio c++ module with reflection metadata.
+
+```cmake
+add_zserio_module(<module-name>
+    ROOT <zserio module source root>
+    ENTRY <entrypoint zserio source. e.G. "main.zs">
+    TOP_LEVEL_PKG <optional: top level package name>)
+```
+
+This will create two CMake targets `<module-name>` and `<module-name>-reflection` (the later links the first).
+To link against the generated zserio module, just link against `<module-name>-reflection`
+
+## Reflection
+
+Zsr stores its top-level reflection metadata in a global list that can be accessed by calling `zsr::packages()`.
+Metadata types and their hiararchy can be found in [runtime/zsr/types.hpp](./runtime/zsr/types.hpp).
+
+There are various utility functions for finding metadata by name: `zsr::find<...>(...)` which take a list of packages to
+search the requested data in.
+See:
+* [runtime/zsr/find.hpp](./runtime/zsr/find.hpp) – Functions for finding and resolving types
+* [runtime/zsr/getset.hpp](./runtime/zsr/getset.hpp) – Functions for getting and setting nested fields
+* [runtime/zsr/introspectable-json.hpp](./runtime/introspectable-json.hpp) – JSON export
+
+Usage example:
+
+```c++
+#include "zsr/reflection-main.hpp"     /* Metadata root */
+#include "zsr/find.hpp"                /* Find utilities */
+#include "zsr/getset.hpp"              /* Get/set utilities */
+#include "zsr/introspectable-json.hpp" /* JSON export */
+
+...
+
+/* Finding a compound named "Test" */
+auto compound = zsr::find<zsr::Compound>(zsr::packages(), "Test");
+
+/* Finding field "a" of compound "A" */
+auto field = zsr::find<zsr::Field>(zsr::packages(), "A.a");
+
+/* Instantiate compound "Test" */
+auto instance = compound->alloc();
```

Comment:
For the sake 🍶 of simplicity, I would prefer `zsr::make` to be used here. I would leave the fact that `zsr::Compound` supports low-level alloc to a different example. Also, I think we should give an example of how initialization parameters can be passed into alloc.